Model/src/main/java/org/gusdb/wdk/model/fix/table/edaanalysis/plugins/VDIMigrationPlugin.java

```diff
@@ -0,0 +1,124 @@
+package org.gusdb.wdk.model.fix.table.edaanalysis.plugins;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.apache.log4j.Logger;
+import org.gusdb.fgputil.json.JsonUtil;
+import org.gusdb.wdk.model.WdkModel;
+import org.gusdb.wdk.model.fix.table.TableRowInterfaces;
+import org.gusdb.wdk.model.fix.table.edaanalysis.AbstractAnalysisUpdater;
+import org.gusdb.wdk.model.fix.table.edaanalysis.AnalysisRow;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class VDIMigrationPlugin extends AbstractAnalysisUpdater {
+  private static final Logger LOG = Logger.getLogger(VDIMigrationPlugin.class);
+  public static final String UD_DATASET_ID_PREFIX = "EDAUD_";
+
+  private Map<String, String> legacyIdToVdiId;
+  private int missingFromVdiCount = 0;
+
+  @Override
+  public TableRowInterfaces.RowResult<AnalysisRow> processRecord(AnalysisRow nextRow) throws Exception {
+    final String legacyDatasetId = nextRow.getDatasetId();
+    final String legacyUdId = legacyDatasetId.replace(UD_DATASET_ID_PREFIX, "");
+    final String vdiId = legacyIdToVdiId.get(legacyUdId);
+
+    if (vdiId == null) {
+      LOG.warn("Unable to find legacy ID " + legacyUdId + " in the tinydb file.");
+      missingFromVdiCount++;
+      return new TableRowInterfaces.RowResult<>(nextRow);
+    }
+
+    // Append UD prefix to VDI ID. The prefix is prepended in the view that maps stable VDI IDs to the unstable study
+    // ID, which is the currency of EDA.
+    final String vdiDatasetId = UD_DATASET_ID_PREFIX + vdiId;
```

Comment:
I thought we were planning to change the previx to VDIID_ or something once we moved to it?  Weren't there multiple hour-long conversations about VD vs VDID vs VDIID?

Reply:
This is just for referring to EDA user datasets that are managed by VDI. I think keeping it as UD should be fine. We discussed in scrum this morning.